server/internal/provider/kaetzchen/cbor_plugins.go

```diff
@@ -136,34 +136,33 @@ func (k *CBORPluginWorker) processKaetzchen(pkt *packet.Packet, pluginClient *cb
 		HasSURB: surb != nil,
 	}
 	cborResponse := <-pluginClient.ReadChan()
-	switch r := cborResponse.(type) {
-	case *cborplugin.Response:
-		if len(r.Payload) > k.geo.UserForwardPayloadLength {
-			// response is probably invalid, so drop it
-			k.log.Errorf("Got response too long: %d > max (%d)",
-				len(r.Payload), k.geo.UserForwardPayloadLength)
-			instrument.KaetzchenRequestsDropped(1)
+	payload, err = cborResponse.Marshal()
+	if err != nil {
+		k.log.Errorf("Failed to Marshal CBOR Plugin response: %v", err)
+		instrument.KaetzchenRequestsDropped(1)
+		return
+	}
+	if len(payload) > k.geo.UserForwardPayloadLength {
+		// response is probably invalid, so drop it
+		k.log.Errorf("Got response too long: %d > max (%d)",
+		len(payload), k.geo.UserForwardPayloadLength)
+		instrument.KaetzchenRequestsDropped(1)
```

Comment:
Since we now have a way to figure out what the max permitted length is (`k.geo.UserForwardPayloadLength`), would it makes sense to add that to the protocol between the provider and the backend service? Like a field passing on `UserForwardPayloadLength: k.geo.UserForwardPayloadLength` so the service has a chance to send a reply to the client saying "sorry I can't give you a response because it would be too long" or whatever, without having to read the KP config?

Reply:
Probably the ServerPlugin method  RegisterConsumer() should be used by client plugins to learn the maximum response sizes. #368